docs/guide/data_migration/importing_data.md

```diff
@@ -0,0 +1,165 @@
+
+``` yaml
+[[= include_file('code_samples/data_migration/examples/create_blog_post_ct.yaml') =]]
+```
+
+### Content
+
+The following example shows how to create two Content items: a folder and an article inside it.
+
+When creating a Content item, two metadata keys are required: `contentType` and `mainTranslation`,
+as well as `parentLocationId`.
+
+To use the Location ID of the folder, which is created automatically by the system,
+you can use a [reference](migration_management.md#references).
+In this case you assign the `parent_folder_location_id` reference name to the Location ID
+and then use it when creating the article.
+
+``` yaml hl_lines="15 24"
+[[= include_file('code_samples/data_migration/examples/create_parent_and_child_content.yaml') =]]
+```
+
+### Roles
+
+The following example shows how to create a Role.
+A Role requires the `identifier` metadata key.
+
+For each Policy assigned to the Role you select the module and function, with optional Limitations.
+
+``` yaml
+[[= include_file('code_samples/data_migration/examples/create_role.yaml') =]]
+```
+
+### Users
+
+The following example shows how to create a user.
+
+The required metadata keys are: `login`, `email`, `password`, `enabled`, `mainLanguage`, and `contentType`.
+You also need to provide the user group's remote content ID.
+
+You can use an [action](data_migration_actions.md) to assign a Role to the user.
+
+``` yaml hl_lines="22-23"
+[[= include_file('code_samples/data_migration/examples/create_user.yaml') =]]
+```
+
+### Language
+
+The following example shows how to create a language.
+
+The required metadata keys are: `languageCode`, `name`, and `enabled`.
+
+``` yaml
+[[= include_file('code_samples/data_migration/examples/create_language.yaml') =]]
+```
+
+### Product catalog
+
+Create an attribute group with two attributes:
+
+``` yaml
+[[= include_file('code_samples/data_migration/examples/create_attribute_group.yaml') =]]
+```
+
+#### Product type
+
+The following example shows how to create a product type.
+
+The main part of the migration file is the same as when creating a regular Content item.
+
+A product type must also contain the definition for an `ibexa_product_specification` Field.
+`fieldSettings` contains information about the product attributes.
+
+``` yaml
+[[= include_file('code_samples/data_migration/examples/create_product_type.yaml') =]]
+```
+
+#### Customer groups
+
+The following example shows how to create a customer group with a defined global price discount:
+
+``` yaml
+[[= include_file('code_samples/data_migration/examples/create_customer_group.yaml') =]]
+```
+
+#### Currencies
+
+The following example shows how to create a currency:
+
+``` yaml
+[[= include_file('code_samples/data_migration/examples/create_currency.yaml') =]]
+```
+
+## Criteria
+
+When using `update` or `delete` modes, you can use criteria to identify the objects to operate on.
```

Comment:
A word of warning: those criteria only work for objects introduced in Product Catalog as of today. Other migrations have their criteria written explicitly, since for them there is no underlying framework that would allow for searching.

Search engine does exist, but has been proven that it's use for migrations might cause data integrity issues (due to how Solr work, especially). Because of this, Content-based migrations do make use of Content filtering, and Product Catalog migrations use filtering that has been baked into all new objects.